From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sing-Han Chen <singhanc@nvidia.com>
Date: Tue, 18 Jun 2024 06:59:42 +0000
Subject: [PATCH 144/240] NVIDIA: SAUCE: usb: xhci: skip mbox configuration

This change skips the mbox configuration for the
project that doesn't need this.

http://nvbugs/4703456

Change-Id: I7acfa239ab389cdc5e264501e202c9b65755fc03
Signed-off-by: Sing-Han Chen <singhanc@nvidia.com>
(cherry picked from commit b344d9876ad025d5900b3411a8d7507e752e8820)
Signed-off-by: Bodla Rakesh Babu <rbodla@nvidia.com>

diff --git a/drivers/usb/host/xhci-tegra.c b/drivers/usb/host/xhci-tegra.c
index 58ed58dfb289..a5ff601e2b40 100644
--- a/drivers/usb/host/xhci-tegra.c
+++ b/drivers/usb/host/xhci-tegra.c
@@ -277,6 +277,7 @@ struct tegra_xusb_soc {
 	bool scale_ss_clock;
 	bool has_ipfs;
 	bool lpm_support;
+	bool skip_mbox_config;
 	bool otg_reset_sspi;
 	bool otg_set_port_power;
 	bool is_xhci_vf;
@@ -1868,7 +1869,7 @@ static int __tegra_xusb_enable_firmware_messages(struct tegra_xusb *tegra)
 	struct tegra_xusb_mbox_msg msg;
 	int err;
 
-	if (tegra->soc->is_xhci_vf)
+	if (tegra->soc->is_xhci_vf || tegra->soc->skip_mbox_config)
 		return 0;
 
 	/* Enable firmware messages from controller. */
@@ -3551,6 +3552,7 @@ static const struct tegra_xusb_soc tegra264_soc = {
 		.owner = XUSB_BAR2_ARU_MBOX_OWNER,
 		.smi_intr = XUSB_BAR2_ARU_SMI_INTR,
 	},
+	.skip_mbox_config = true,
 	.lpm_support = true,
 	.has_bar2 = true,
 	.has_pg_support = true,
-- 
2.55.0

